packages/app/src/components/routes.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useEffect } from 'react';
+import { Home } from '../pages/home';
+import { Create, SaveKey } from '../pages/sign-up';
+import { SignIn, DecryptRecoveryCode } from '../pages/sign-in';
+
+import { Username } from '../pages/username';
+import { UsernameRegistryError } from '../pages/registery-error';
+import { SecretKey } from '../pages/secret-key';
+
+import { ChooseAccount } from '../pages/connect';
+
+import { doSaveAuthRequest } from '@store/onboarding/actions';
+import { useDispatch } from 'react-redux';
+import { ScreenPaths } from '@store/onboarding/types';
+import { doFinishSignIn as finishSignIn } from '@store/onboarding/actions';
+import { authenticationInit } from '@common/utils';
+import { useAnalytics } from '@common/hooks/use-analytics';
+import { useWallet } from '@common/hooks/use-wallet';
+import { useOnboardingState } from '@common/hooks/use-onboarding-state';
+
+import { Routes as RoutesDom, Route, Navigate } from 'react-router-dom';
+
+export const Routes: React.FC = () => {
+  const dispatch = useDispatch();
+  const { doChangeScreen } = useAnalytics();
+  const { identities } = useWallet();
+  const { isOnboardingInProgress } = useOnboardingState();
+  const authRequest = authenticationInit();
+
+  useEffect(() => {
+    if (authRequest) {
+      dispatch(doSaveAuthRequest(authRequest));
+    }
+  }, [authRequest]);
+
+  const doFinishSignIn = ({ identityIndex }: { identityIndex: number } = { identityIndex: 0 }) =>
+    dispatch(finishSignIn({ identityIndex }));
+
+  const isSignedIn = !isOnboardingInProgress && identities.length;
+
+  return (
+    <RoutesDom>
```

Comment:
Because this is using "real URLs", and not using a hash-based router, this isn't working in production. Even if we got netlify setup for this, the extension won't be able to ever support this kind of functionality. I think we can just use `HashRouter` from `react-router-dom` here, and it'll work everywhere.